### PR TITLE
Add tinycore package

### LINKED ISSUE #70: Support tinycore packages
To support tinycore 64 bit:

Create a glibc wrapper.
/usr/local/bin/wrap:

``` sh
LD_LIBRARY_PATH=/tmp/image/usr/local/lib /lib64/ld-linux-x86-64.so.2 /tmp/image/usr/local/bin/$@
```

Grab a tinycore package, like binutils:

``` sh
cd ~/Downloads;
curl -O http://distro.ibiblio.org/tinycorelinux/6.x/x86_64/tcz/binutils.tcz
sudo mount -o loop binutils.tcz /tmp/image
```

Now you can run it with your wrapper:

``` sh
wrap ld
```

Tinycore (6.x) binutils is about 2.5megs vs the 37meg chromebrew binutils; chromebrew decompresses to 100megs whereas tcz files are mounted via loopback. Feels like a big win.

-Charles

### PR DESCRIPTION
**Pull request by JL2210**
Fixes #70

## Description
It starts with a recent Linux kernel, vmlinuz, and our root filesystem and start-up scripts packaged with a basic set of kernel modules in core.gz. Core (11MB) is simply the kernel + core.gz - this is the foundation for user created desktops, servers, or appliances. TinyCore is Core + Xvesa.tcz + Xprogs.tcz + aterm.tcz + fltk-1.3.tcz + flwm.tcz + wbar.tcz

---

Thoroughly tested and working on:
- [x] x86_64
- [x] i686
- [ ] armv7l, aarch64

### CODE REVIEW

File: packages/tinycore.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+set -e
+
+export PKG=`echo \$\{1%%.*\}`
+export BIN=`echo $1|sed \"s/^[^\.]*\.//\"`
+export OPT=`shift; echo $@`
+
+[ -d #{CREW_PREFIX}/share/tinycore/\$\PKG ] || { echo Package $PKG not found. && exit 1; }
+[ -e #{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/bin/\$\BIN ] || { echo Program $BIN not found. && exit 1; }
+LD_LIBRARY_PATH=\$\LD_LIBRARY_PATH:#{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/lib #{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/bin/\$\BIN $OPT
```
Comment (uberhacker) [inline packages/tinycore.rb]: Looks like you got carried away with the backslashes.  You shouldn't need those after each `$`.

File: packages/tinycore.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    puts "To try TinyCore Bash, please execute:".lightblue
+    puts "sudo mount -o loop #{CREW_PREFIX}/share/tinycore/examples/bash.tcz #{CREW_PREFIX}/share/tinycore/bash/".lightblue
+    puts "tinycore bash".lightblue
+    puts "sudo umount #{CREW_PREFIX}/share/tinycore/bash/".lightblue
+    puts
+    puts "The executable names are in the format of Snap executable names.".lightblue
+    puts "For eample, to execute the \'bashbug\' program in the TinyCore \'bash\' package, please execute:".lightblue
+    puts "tinycore bash.bashbug".lightblue
+    puts
+    puts "To run a binary that has a dot in its name, such as \'ld.gold\', use the full name:".lightblue
```
Comment (uberhacker) [inline packages/tinycore.rb]: The single quotes do not need escaped.

File: packages/tinycore.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  end
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  def self.build
+    case ARCH
+    when 'x86_64'
```
Comment (uberhacker) [inline packages/tinycore.rb]: What about `i686` and `armv7l/aarch64`?

File: packages/tinycore.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+export PKG=`echo \$\{1%%.*\}`
+export BIN=`echo $1|sed \"s/^[^\.]*\.//\"`
+export OPT=`shift; echo $@`
+
+[ -d #{CREW_PREFIX}/share/tinycore/\$\PKG ] || { echo Package $PKG not found. && exit 1; }
+[ -e #{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/bin/\$\BIN ] || { echo Program $BIN not found. && exit 1; }
+if file #{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/bin/\$\BIN | grep \"text executable\" &>/dev/null; then
+  LD_LIBRARY_PATH=\$\LD_LIBRARY_PATH:#{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/lib #{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/bin/\$\BIN $OPT
+else
+  LD_LIBRARY_PATH=\$\LD_LIBRARY_PATH:#{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/lib /#{ARCH_LIB}/ld-linux-x86-64.so.2 #{CREW_PREFIX}/share/tinycore/\$\PKG/usr/local/bin/\$\BIN $OPT
```
Comment (uberhacker) [inline packages/tinycore.rb]: You should try and abstract the `x86-64` also.

File: packages/tinycore.rb (hunk 2 above)
Comment (JL2210) [reply packages/tinycore.rb]: Got it. Will fix tomorrow.

File: packages/tinycore.rb (hunk 1 above)
Comment (JL2210) [reply packages/tinycore.rb]: OK.

File: packages/tinycore.rb (hunk 3 above)
Comment (JL2210) [reply packages/tinycore.rb]: Halfway down, it says `else`. Read.

File: packages/tinycore.rb (hunk 4 above)
Comment (JL2210) [reply packages/tinycore.rb]: ARCH is x86_64. It's good.

File: packages/tinycore.rb (hunk 3 above)
Comment (uberhacker) [reply packages/tinycore.rb]: Gotcha, missed the else.

**Comment by JL2210** [comment]
How is this?

**Comment by JL2210** [comment]
GitHub web editing isn't half bad. I like the nano+vim approach better, though.

**Comment by JL2210** [comment]
Nano to do the real editing, and vim for mass-deletion of lines.

**Comment by uberhacker** [comment]
vim minus nano would be even better. :)

**Comment by uberhacker** [comment]
I would argue you can't do real editing outside of vim.

**Comment by JL2210** [comment]
Yeah, but Nano is easy. I used to be a big vim guy, but I forgot.

**Pull request merged**